sdk/dapp-kit/src/hooks/useSuiClientMutation.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { UseMutationOptions } from '@tanstack/react-query';
+import { useMutation } from '@tanstack/react-query';
+import { useSuiClientContext } from './useSuiClient.js';
+import type { SuiRpcMethods } from './useSuiClientQuery.js';
+
+export type UseSuiClientMutationOptions<T extends keyof SuiRpcMethods> = Omit<
+	UseMutationOptions<SuiRpcMethods[T]['result'], unknown, SuiRpcMethods[T]['result'], unknown[]>,
+	'mutationFn'
+>;
+
+export function useSuiClientMutation<T extends keyof SuiRpcMethods>(
+	{
+		method,
+		params,
+	}: {
+		method: T;
+		params: SuiRpcMethods[T]['params'];
+	},
+	options: UseSuiClientMutationOptions<T> = {},
+) {
+	const suiContext = useSuiClientContext();
+
+	return useMutation({
+		...options,
```

Comment:
should we specify a mutation key like we do for `useSuiClientQuery`?

Reply:
this is where my lack of front end experience shows through... what is that used for?  For queries it's for caching, does that make sense for mutations?